app/src/main/java/org/oppia/android/app/spotlight/SpotlightFragment.kt

```diff
@@ -0,0 +1,430 @@
+    activity.windowManager.defaultDisplay.getMetrics(displayMetrics)
+
+    screenHeight = displayMetrics.heightPixels
+    screenWidth = displayMetrics.widthPixels
+  }
+
+  // since this fragment does not have any view to inflate yet, all the tasks should be done here.
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    (fragmentComponent as FragmentComponentImpl).inject(this)
+
+    internalProfileId = arguments?.getInt(PROFILE_ID_ARGUMENT_KEY) ?: -1
+    calculateScreenSize()
+  }
+
+  /**
+   * Requests a spotlight to be shown on the [SpotlightTarget]. The spotlight is enqueued if it
+   * [SpotlightTarget] hasn't been shown before in a FIFO buffer. This API can ensure proper
+   * spotlighting of a which is laid out after an animation. For the spotlights to work correctly,
+   * we must know the [SpotlightTarget]'s anchor's size and position. For a view that is laid out
+   * after an animation, we must wait until the final size and positions of the anchor view are
+   * measured, which can be achieved by using doOnPreDraw. Refer:
+   * (https://betterprogramming.pub/stop-using-post-postdelayed-in-your-android-views-9d1c8eeaadf2)
+   *
+   * @param spotlightTarget The [SpotlightTarget] for which the spotlight is requested
+   */
+  override fun requestSpotlightViewWithDelayedLayout(spotlightTarget: SpotlightTarget) {
+    spotlightTarget.anchor.doOnPreDraw {
+      if (it.visibility != View.VISIBLE) {
+        return@doOnPreDraw
+      }
+      val targetAfterViewFullyDrawn = SpotlightTarget(
+        it,
+        spotlightTarget.hint,
+        spotlightTarget.shape,
+        spotlightTarget.feature
+      )
+      requestSpotlight(targetAfterViewFullyDrawn)
+    }
+    spotlightTarget.anchor.invalidate()
+    spotlightTarget.anchor.requestLayout()
+  }
+
+  /**
+   * Requests a spotlight to be shown on the [SpotlightTarget]. The spotlight is enqueued if it
+   * hasn't been shown before in a FIFO buffer. This API can ensure proper spotlighting of a
+   * [SpotlightTarget] when it is laid out late due to a data provider call. It cannot ensure the
+   * same if the view has to be spotlight immediately after an animation. It also cannot spotlight
+   * targets which are a part of a recycler view which are laid out after a data provider call. If
+   * TalkBack is turned on, no spotlight shall be shown.
+   *
+   * @param spotlightTarget The [SpotlightTarget] for which the spotlight is requested
+   */
+  override fun requestSpotlight(spotlightTarget: SpotlightTarget) {
+
+    if (accessibilityService.isScreenReaderEnabled()) return
+
+    val profileId = ProfileId.newBuilder()
+      .setInternalId(internalProfileId)
+      .build()
+
+    val featureViewStateLiveData =
+      spotlightStateController.retrieveSpotlightViewState(
+        profileId,
+        spotlightTarget.feature
+      ).toLiveData()
+
+    // use activity as observer because this fragment's view hasn't been created yet.
```

Comment:
```suggestion
    // Use activity as observer because this fragment's view hasn't been created yet.
```

For consistency. Ditto elsewhere in the PR.

Reply:
done